pkg/operator/controller/extension/runtimecluster.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extension
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+func runtimeClusterAdmissionManagedResourceName(extension *operatorv1alpha1.Extension) string {
+	return fmt.Sprintf("extension-admission-runtime-%s", extension.Name)
```

Comment:
Do we need to pay attention to name length restrictions in case the extension name gets longer?

Reply:
As discussed, name limit is at 253 characters.